app/components/provider_interface/application_timeline_component.rb

```diff
@@ -24,15 +26,22 @@ def initialize(application_choice:)
 
   private
 
+    def with_activity_log_events_for(attr)
+      filtered = @activity_log_events.select { |event| event.changes.key?(attr) }
+      filtered.map { |event| yield event }
```

Comment:
What does the `yield` on this line do?

Reply:
Yields once per event, it's what enables
```
      with_activity_log_events_for('status') do |event|
        Event.new(
```
etc. below.

It abstracts away the filtering, while still letting you control the construction of Event structs.

Reply:
Ah I see, missed that we weren't doing a `map` :)

Reply:
Yes, I guess it's easy to miss. Renaming the method to `map_activity_log_events_for`.